extensions/lucide-icons/src/index.tsx

```diff
@@ -1,45 +1,101 @@
-import { Action, ActionPanel, Color, environment, Grid, Icon } from "@raycast/api";
-import { useCachedState } from "@raycast/utils";
-import { readdirSync, readFileSync } from "fs";
-import { join } from "path";
-import { useMemo } from "react";
+import { Action, ActionPanel, Color, Grid, Icon } from "@raycast/api";
+import { useCachedPromise, useCachedState } from "@raycast/utils";
+import fetch from "node-fetch";
 
 // Lucide's toPascalCase function
 export const toPascalCase = (string: string) => {
-  const camelCase = string.replace(/^([A-Z])|[\s-_]+(\w)/g, (match, p1, p2) =>
-    p2 ? p2.toUpperCase() : p1.toLowerCase()
+  const camelCase = string.replace(/^([A-Z])|[\s-_]+(\w)/g, (_match, p1, p2) =>
+    p2 ? p2.toUpperCase() : p1.toLowerCase(),
   );
 
   return camelCase.charAt(0).toUpperCase() + camelCase.slice(1);
 };
 
+type LucideIcon = {
+  name: string;
+  content: string;
+  path: string;
+  component: string;
+};
+
 export default function Command() {
-  const files = useMemo(
-    () =>
-      readdirSync(join(environment.assetsPath, "icons"))
-        .filter((file) => file.endsWith(".svg"))
-        .map((file) => {
-          return [join(environment.assetsPath, "icons", file), file.split(".")[0]];
-        }),
-    []
+  const { data, isLoading } = useCachedPromise(
+    async (iconNodesUrl: string) => {
+      const response = await fetch(iconNodesUrl)
+        .then((res) => res.json() as Promise<Record<string, unknown>>)
+        .then((data) => Object.keys(data));
+
+      const promises = response.map(async (iconName) => {
```

Comment:
How long does it take to fetch all of the icons? The SVG data is already present, albeit in a parsed form, so it might be better to assemble the data into SVGs without refetching them again.

Reply:
is there a way to rerun this bit of code that updates them every once in a while?

Reply:
I don't know of any native ways, but I haven't touched the Raycast developer ecosystem in a while.

Reply:
You can save a datetime object and compare - could that work?

Reply:
thought about saving it on the raycast cache, makes sense.